### ISSUE TITLE
Cannot select exit node on Android Beta

### ISSUE THREAD

**Issue by anaisbetts** [tailscale/tailscale#11715]
### What is the issue?

In the new Tailscale Android beta, I cannot select any Exit Node, the list is empty despite being both a Mullvad subscriber as well as having several of my own exit nodes

### OS

Android

### OS version

Android 14 (Samsung Z-Flip 5, One UI 6.1)

### Tailscale version

v1.65.0-t220764331-g50bb25aca2d

### Bug report

BUG-9980889fc0a0502848710fa3714585cc0a38d5232b0baa4116f188bd777f0b9c-20240413080226Z-bdfa018220dfa6e5

**Comment by anaisbetts** [comment]
![SmartSelect_20240413_100324](https://github.com/tailscale/tailscale/assets/1396/264b4e09-5ca0-4afc-9b75-0851f2a03344)

**Comment by agottardo** [comment]
Do your exit nodes appear in the devices list instead (the main screen of the app)?

**Comment by anaisbetts** [comment]
@agottardo No, other than the normal appearance of a machine in the list (i.e. one of my machines is an Exit Node, but it's there similar to all the other Tailscale machines, clicking it just shows the standard information)

**Comment by agottardo** [comment]
It sounds like the devices are not getting sent down from the coordination server to the client then. Did you set up the Android device as a new device? Do your ACLs allow it to talk to the other devices in your network?

**Comment by kelivel** [comment]
@anaisbetts I took a quick look at this and your self-hosted exit node is currently unapproved, so that device won't appear in your app as a working exit node. Also make sure you've granted [a Mullvad license](https://login.tailscale.com/admin/settings/general/mullvad) to your Android device in the admin panel. Let us know if you still can't connect to your exit nodes afterwards and we can take a closer look.

**Comment by anaisbetts** [comment]
@kelivel You're right, fixing this solved everything - here's what happened:

1. Somewhere along the line I nerfed my original exit node, probably when enabling Tailscale SSH
2. Because of Google Play Beta Account Nonsense:tm:, I had to uninstall Tailscale and reinstall it. When I did that, my old phone name was lost and Tailscale made a new one - I didn't think about it and just delete + renamed it. However, only the old phone "machine" was in the Mullvad approved list, and the new one ofc was not.

Everything works now so I'll close this out, thanks so much for having a look!
